src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -541,14 +543,25 @@ public function compile()
         $compiler->compile($this);
         $this->compiled = true;
 
+        $renameDefinitions = array();
         foreach ($this->definitions as $id => $definition) {
             if (!$definition->isPublic()) {
-                $this->privates[$id] = true;
+                if (null === $privateOriginId = $definition->getPrivateOriginId()) {
```

Comment:
Yeah about this.. i intended this as a fallback scenario when there's no randomizing compiler pass. 

So if randomizing is bypassed, how do we deal with privates? Track them as before?